### ISSUE TITLE
YouTube/YouTube Music: Wrong video IDs fetched for YouTube Music (IDs are unavailable on YouTube)

### ISSUE THREAD

**Issue by Endothermic-Dragon** [yt-dlp/yt-dlp#10745]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

United States

### Provide a description that is worded well enough to be understood

# What the problem is
In certain playlists in YouTube Music, `yt-dlp` sometimes fetches the wrong video ID. For instance, I've compiled three such examples in a playlist below.
https://music.youtube.com/playlist?list=PL2--NoRqqdncIQnFp4eTFMLFLybr4Tb5v

When I view this playlist on YouTube Music, it gives me three playable videos.

![image](https://github.com/user-attachments/assets/fa6e7443-9c75-411b-abc4-19382708b431)

When I try to play them, YouTube Music plays them with the following IDs:
- `u8ibz4-IbjM`
- `TYK1xqcAhQE`
- `PwspPS0v-WE`

These three videos are playable on both YouTube and YouTube Music.

But, when I try and fetch the video IDs with `yt-dlp`, i get the following IDs:
- `BfDaZ2wLDVA`
- `rzDimcqx_vs`
- `d5ATqtjEMhY`

These three videos are unavailable on both YouTube and YouTube Music.

Additionally, I'm not sure if this is important, but the same playlist viewed on YouTube says that the three videos are unavailable.
https://www.youtube.com/playlist?list=PL2--NoRqqdncIQnFp4eTFMLFLybr4Tb5v

![image](https://github.com/user-attachments/assets/b25344c4-dd0e-4e5d-a4c3-910957cacdaf)

TLDR: `yt-dlp` occasionally reports incorrect video IDs on YouTube Music playlists.

# How it could be fixed
I tried reading the code so I could fix it myself. After some poking around, I reached my conclusion:
¯\\\_(ツ)\_/¯
(This is well beyond my skill level)

# How your proposed solution would look like
Worst-case scenario, perform a `ytsearch10` and look for exact song name + artist name matches. Return the first one.

Also, I noticed that sometimes YouTube likes to append `" - Topic"` to the artist name, so it shouldn't be too hard to detect and remove that (after fetching the artist name, of course).

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '--flat-playlist', '--print', 'id', 'https://music.youtube.com/playlist?list=PL2--NoRqqdncIQnFp4eTFMLFLybr4Tb5v']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.08.06 from yt-dlp/yt-dlp [4d9231208] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.22631-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 6.1.1-full_build-www.gyan.dev (setts), ffprobe 6.1.1-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.07.04, curl_cffi-0.5.10, mutagen-1.47.0, requests-2.32.3, sqlite3-3.35.5, urllib3-2.2.2, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1830 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.08.06 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.08.06 from yt-dlp/yt-dlp)
[youtube:tab] Extracting URL: https://music.youtube.com/playlist?list=PL2--NoRqqdncIQnFp4eTFMLFLybr4Tb5v
WARNING: [youtube:tab] YouTube Music is not directly supported. Redirecting to https://www.youtube.com/playlist?list=PL2--NoRqqdncIQnFp4eTFMLFLybr4Tb5v
[youtube:tab] PL2--NoRqqdncIQnFp4eTFMLFLybr4Tb5v: Downloading webpage
WARNING: [youtube:tab] YouTube said: INFO - 3 unavailable videos are hidden
[youtube:tab] PL2--NoRqqdncIQnFp4eTFMLFLybr4Tb5v: Redownloading playlist API JSON with unavailable videos
[download] Downloading playlist: yt-dlp Issue Report (Unlisted Playlist)
[youtube:tab] PL2--NoRqqdncIQnFp4eTFMLFLybr4Tb5v page 1: Downloading API JSON
[youtube:tab] Playlist yt-dlp Issue Report (Unlisted Playlist): Downloading 3 items of 3
[debug] The information of all playlist entries will be held in memory
[download] Downloading item 1 of 3
BfDaZ2wLDVA
[download] Downloading item 2 of 3
rzDimcqx_vs
[download] Downloading item 3 of 3
d5ATqtjEMhY
[download] Finished downloading playlist: yt-dlp Issue Report (Unlisted Playlist)
```

**Comment by bashonly** [comment]
Duplicate of #6351, #10695
